compiler/z/codegen/BinaryEvaluator.cpp

```diff
@@ -2466,8 +2466,36 @@ genericRotateAndInsertHelper(TR::Node * node, TR::CodeGenerator * cg)
    return NULL;
    }
 
+/*
+ *
+ * This is a helper function that will try to use a rotate instruction
+ * to perform a 'land' operation. For example, if we have the following tree:
+ *
+ * land
+ *    i2l
+ *       iload parm=0
+ *    lconst 0xFFFFFFFFF0000000
+ *
+ * We can avoid the 'land' operation altogether in such a case. We know that
+ * the result of this operation will preserve the first 16 bits of the iload
+ * parameter, and zero out the rest. This can be accomplished by using a RISBG 
+ * instruction where we specify the start and end position of the bits to preserve,
+ * specifying a rotate factor of 0 (we don't want to move the location of the
+ * preserved bits), and setting the zero bit which will tell the CPU to zero out
+ * all bits except for the ones in the preserved range. 
+ *
+ * ex. If in the above tree the value of our iload parameter is 
+ * 0x80000000. This will be converted to a long and become
+ * 0xFFFFFFFF80000000. Traditionally we would need to use two
+ * AND instructions to capture the correct result (we need to 
+ * AND the top half and bottom half of the word separately using 
+ * NIHF and NILF instructions). However, with RISBG, we can just choose
+ * to preserve the first 36 bits (lconst's first 36 bits are 1's) and 
+ * accomplish this in one instruction.  
+ *
+ */
```

Comment:
Can we properly Doxygen this function an its parameters in the header and leave the details Doxygen'd in the object file implementation?